method.rb

```diff
@@ -0,0 +1,113 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    output_array = is_a?(Array) ? self : to_a
+    i = 0
+    while i < size
+      yield output_array[i]
+      i += 1
+    end
+    output_array
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    output_array = is_a?(Array) ? self : to_a
+    i = 0
+    while i < size
+      yield output_array[i], i
+      i += 1
+    end
+    output_array
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    output_array = []
+    my_each do |item|
+      output_array << item if (yield item) == true
+    end
+    output_array
+  end
+
+  def my_all?(parameter = nil) # rubocop:disable Metrics/CyclomaticComplexity
+    status_cumulator = true
+    i = 0
+
+    loop do
+      status_cumulator &&= yield self[i] if block_given?
+      status_cumulator &&= !parameter.match(self[i]).nil? if parameter.is_a?(Regexp)
+      status_cumulator &&= self[i].is_a(parameter) if parameter.is_a?(Class)
+      status_cumulator &&= (self[i] == parameter) if parameter
+      i += 1
+      break if i == size || status_cumulator == false
+    end
+    status_cumulator
+  end
+
+  def my_any?(parameter = nil) # rubocop:disable Metrics/CyclomaticComplexity
+    status_cumulator = false
+    i = 0
+    loop do
+      status_cumulator ||= yield self[i] if block_given?
+      status_cumulator ||= !parameter.match(self[i]).nil? if parameter.is_a?(Regexp)
+      status_cumulator ||= self[i].is_a(parameter) if parameter.is_a?(Class)
+      status_cumulator ||= (self[i] == parameter) if parameter
+      i += 1
+      break if i == size || status_cumulator == true
+    end
+    status_cumulator
+  end
+
+  def my_none?(parameter = nil)
+    !my_any?(parameter)
```

Comment:
* [ ] should be a working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
#### compare the following examples by replacing the custom method with the original method in each example and rectify the error that arises.
```p %w{ant bear cat}.my_none? { |word| word.length >= 4 } # must return false```

```p [ ].my_none? # must return true```

```p [nil, false, true].my_none? # must return false```